# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/test/mochitest/test_notification_worker_click.html
# Commit: 11a3c2c86a8e
# Full Hash: 11a3c2c86a8e85fbbfe0fad4d36bf3956af109c5
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 9: Implement alertclickcallback behavior r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227694
# ==============================================================================

diff -r 10bb5f41f44b -r 11a3c2c86a8e dom/notification/test/mochitest/test_notification_worker_click.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/notification/test/mochitest/test_notification_worker_click.html	Mon Nov 11 20:34:14 2024 +0000
@@ -0,0 +1,28 @@
+<!DOCTYPE HTML>
+<meta charset="utf-8">
+<script src="/tests/SimpleTest/SimpleTest.js"></script>
+<script src="/tests/dom/notification/test/mochitest/MockAlertsService.js"></script>
+<script src="/tests/dom/notification/test/mochitest/NotificationTest.js"></script>
+<link rel="stylesheet" href="/tests/SimpleTest/test.css" />
+
+<script>
+add_task(async function test_notification_worker_click() {
+  await NotificationTest.allowNotifications();
+  await MockAlertsService.register();
+  await MockAlertsService.enableAutoClick();
+
+  var w = new Worker(URL.createObjectURL(new Blob([`
+    new Notification("click").onclick = () => {
+      postMessage("clicked");
+    };
+  `])));
+  await new Promise(resolve => {
+    w.onmessage = function(e) {
+      if (e.data === 'clicked') {
+        resolve();
+      }
+    }
+  });
+  ok(true, "notification click event happened from worker");
+});
+</script>
